classes/db/Db.php

```diff
@@ -567,10 +567,10 @@ public function executeS($sql, $array = true, $use_cache = true)
 		if ($use_cache && $this->is_cache_enabled && $array)
 		{
 			$this->last_query_hash = Tools::encryptIV($sql);
-			if (($result = Cache::getInstance()->get($this->last_query_hash)) !== false)
+			if (Cache::getInstance()->exists($this->last_query_hash) !== false)
```

Comment:
Hello,
I agree with @letwang point because it may introduce a bug, and also because the code is more expressive this way.
If you want to avoid multiple calls, maybe you should consider using the Optional pattern ? https://github.com/typedphp/optional

@jocel1 could you explain your point ? As FS and APC/Memcached are both implementations of the Cache abstraction, shouldn't they be replaceable without any side effect ?

Last but not least : I'm sure that some unit or integration testing would highlight the real behaviour of this code.